test/performance/latency/LatencyTestSubscriber.cpp

```diff
@@ -461,50 +588,80 @@ bool LatencyTestSubscriber::test(
     }
     else
     {
-        latency_type_ = new LatencyType(datasize);
+        latency_type_ = static_cast<LatencyType*>(latency_data_type_->createData());
+        latency_type_->data.resize(datasize, 0);
     }
 
-    std::unique_lock<std::mutex> lock(mutex_);
-    if (command_msg_count_ == 0)
-    {
-        command_msg_cv_.wait(lock);
-    }
-    --command_msg_count_;
+    // Wait for the Publisher READY command
+    unique_lock<mutex> lock(mutex_);
+    command_msg_cv_.wait(
+        lock,
+        [this]()
+        {
+            return command_msg_count_ != 0;
+        });
+    command_msg_count_ = 0;
     lock.unlock();
 
+    // Send to Publisher the BEGIN command
     test_status_ = 0;
     received_ = 0;
     TestCommandType command;
     command.m_command = BEGIN;
-    std::cout << "Testing with data size: " << datasize + 4 << std::endl;
-    command_publisher_->write(&command);
+    if (!command_writer_->write(&command))
+    {
+        logError(LatencyTest, "Subscriber fail to publish the BEGIN command")
+        return false;
+    }
+
+    logInfo(LatencyTest, "Testing with data size: " << datasize + 4);
 
+    // Wait for the STOP or STOP_ERROR commands
     lock.lock();
-    command_msg_cv_.wait(lock, [&]()
-            {
-                return command_msg_count_ > 0;
-            });
-    --command_msg_count_;
+    command_msg_cv_.wait(
+        lock,
+        [this]()
+        {
+            return command_msg_count_ != 0;
+        });
+    command_msg_count_ = 0;
     lock.unlock();
 
-    std::cout << "TEST OF SIZE: " << datasize + 4 << " ENDS" << std::endl;
-    std::this_thread::sleep_for(std::chrono::milliseconds(50));
+    logInfo(LatencyTest, "TEST OF SIZE: " << datasize + 4 << " ENDS");
+    this_thread::sleep_for(chrono::milliseconds(50));
 
     size_t removed;
-    this->data_publisher_->removeAllChange(&removed);
+    data_writer_->clear_history(&removed);
 
     if (dynamic_data_)
     {
-        DynamicDataFactory::get_instance()->delete_data(dynamic_data_type_);
+        dynamic_pub_sub_type_->deleteData(dynamic_data_type_);
     }
     else
     {
-        delete(latency_type_);
+        latency_data_type_->deleteData(latency_type_);
```

Comment:
I think the original code was ok.

Reply:
If we create the data using TypeSupport we must stick to it in destruction to match the allocator used.